src/cli.ts

```diff
@@ -182,3 +189,44 @@ function prepareCliMessage(
 
 	return spaced ? `\n${preparedMessage}\n` : preparedMessage;
 }
+
+export function printEnvInfo(): void {
+	cliLog(`
+		System:
+			Platform: ${os.platform()}
+			Arch: ${os.arch()}
+			Version: ${os.version()}
+			CPU: (${os.cpus().length}) ${os.arch()} ${os.cpus()[0]?.model}
+			Memory: ${Math.round(os.totalmem() / 1024 / 1024 / 1024)} GB
+			Shell: ${process.env['SHELL']?.toString() ?? 'Unknown'}
+		Binaries:
+			Node: ${process.versions.node}
+			Yarn: ${getBinaryVersion('yarn')}
+			npm: ${getBinaryVersion('npm')}
+			pnpm: ${getBinaryVersion('pnpm')}
+		Relevant Packages:
+			vercel: ${getPackageVersion('vercel')}
+			next: ${getPackageVersion('next')}
+			react: ${getPackageVersion('react')}
+			react-dom: ${getPackageVersion('react-dom')}
+			eslint-config-next: ${getPackageVersion('eslint-config-next')}
+	`);
+}
+
+function getPackageVersion(packageName: string): string {
+	try {
+		// eslint-disable-next-line @typescript-eslint/no-var-requires
+		return require(`${packageName}/package.json`).version;
+	} catch {
```

Comment:
@dario-piotrowicz, I am wondering whether we should use the package.json from project root, which would ensure that the package version is always available, even if the package is not installed or if we should require package.json for each package. While requiring package.json for each package works fine locally, I am encountering difficulties in figuring out why it is not working with prerelease.

Reply:
@tarunrajput I don't think we can use the `package.json` from the project root, there you often specify packages using `^` or `~` so we could know the generic version there but not what the specific version the project is actually using (if this is what you were suggesting).

You could try using the `package-lock.json` file but that might be complicated, and you'd also have to deal with other type of lock files :sweat: 

one alternative I think is trying to use `npm list --json` as that outputs a json which should give you all the info you need (but only for installed packages, which might be fine?):
![Screenshot at 2023-04-23 15-07-42](https://user-images.githubusercontent.com/61631103/233844624-cdf5b9fd-173b-445a-8a00-05e69e541ae1.png)

___

By the way, your prerelease does look to be working fine for me:
![Screenshot at 2023-04-23 14-43-08](https://user-images.githubusercontent.com/61631103/233844721-dcb5b6d1-ce81-44a9-9470-8e5b0675d85d.png)

The only issue I can see if that if I link a project locally with a local version of `next-on-pages` the process is unable to get the version of the various packages since the working directory for the `next-on-pages` package is different from that of the current project (which is not ideal, my solution above should hopefully handle that correctly, although it's not a very common/important use case so if we can't get that to work I don't think it would be a deal breaker)

Reply:
(PS: with `npm list` we would also get rid of that ugly `require` :smiley:)